src/syncerd/runtime.rs

```diff
@@ -104,12 +107,18 @@ impl esb::Handler<ServiceBus> for Runtime {
         endpoints: &mut Endpoints,
         bus: ServiceBus,
         source: ServiceId,
-        request: Request,
+        request: BusMsg,
     ) -> Result<(), Self::Error> {
-        match bus {
-            ServiceBus::Msg => self.handle_rpc_msg(endpoints, source, request),
-            ServiceBus::Ctl => self.handle_rpc_ctl(endpoints, source, request),
-            ServiceBus::Bridge => self.handle_bridge(endpoints, source, request),
+        match (bus, request) {
+            (ServiceBus::Msg, request) => self.handle_msg(endpoints, source, request),
+            // Control bus for internal command
+            (ServiceBus::Ctl, BusMsg::Ctl(req)) => self.handle_ctl(endpoints, source, req),
+            // User issued command RPC bus, only accept BusMsg::Rpc
+            (ServiceBus::Rpc, BusMsg::Rpc(req)) => self.handle_rpc(endpoints, source, req),
+            // Syncer event bus
```

Comment:
comment here that this is not consistent with the diagram in
https://github.com/farcaster-project/farcaster-node/issues/706#issue-1355968942.
I'm totally ok with syncerd talking sending info via RPC bus - just making sure
that this is intentional. Would be nice if we have a diagram that's consistent
with the changes in this PR in the docs (likewise accounting for the fact that
cli sends instructions via CTL bus).

Reply:
Yes the diagram must be updated, and the refactor also contain some bus handling (that are empty) that will be removed while simplifying the Hello msg at the start.

Reply:
I'll post an update on this PR before moving it to the diagram folder in the source code.